### PR TITLE
i#3479: Move RUN_IN_RELEASE labeling to test lists.

### LINKED ISSUE #3479: Run at least a few release-build tests on Travis
In light of a serious regression crashing all drcachesim/drmemtrace uses in #3478 which was not detected pre-commit but in our post-commit CDash tests, this issue proposes adding at least some subset of tests to be run in release build on Travis.  The reason we don't run all of them is to keep the pre-commit time down to something reasonable.

I'm proposing giving a few key tests "TRAVIS" labels and run with "-L TRAVIS".  We're already doing the release build on Travis so a few tests won't add too much time.

For reference on how long it takes to build vs run tests, here are the numbers on my CDash Linux machine:

debug-internal-64
build: 1:54
tests: 10:28

release-external-64
build: 1:42
tests: 6:13

### PR DESCRIPTION
**Pull request by hgreving2304**
Moves the assignment of RUN_IN_RELEASE label to the "torun" function in order to make
it independent from architecture and other flags. 'drcachesim', 'drcacheoff' and 'common'
tests have been added. This is a slight increase over the previous RUN_IN_RELEASE list,
as now all 'common' tests are part of this label.

Fixes #3479

### CODE REVIEW

File: suite/tests/CMakeLists.txt [hunk 1]
```diff
@@ -1459,6 +1459,18 @@ function(torun test key source native standalone_dr dr_ops exe_ops added_out)
     # global reg keys and can thus run multiple instances of the same
     # app name simultaneously so we do not need to mark RUN_SERIAL
   endif (WIN32)
+
+  # A subset of release tests has been added and labeled with RUN_IN_RELEASE. It focuses
+  # on Google's use of the drcachesim client. We do not want to extend the Travis
+  # regression time by too much and keep the following list concise.
+  # amhere
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: Mistakenly left: please remove.

File: suite/tests/CMakeLists.txt [hunk 2]
```diff
@@ -1459,6 +1459,18 @@ function(torun test key source native standalone_dr dr_ops exe_ops added_out)
     # global reg keys and can thus run multiple instances of the same
     # app name simultaneously so we do not need to mark RUN_SERIAL
   endif (WIN32)
+
+  # A subset of release tests has been added and labeled with RUN_IN_RELEASE. It focuses
+  # on Google's use of the drcachesim client. We do not want to extend the Travis
+  # regression time by too much and keep the following list concise.
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: What is the time increase with this new list?

File: suite/tests/CMakeLists.txt [hunk 3]
```diff
@@ -1459,6 +1459,18 @@ function(torun test key source native standalone_dr dr_ops exe_ops added_out)
...
+  # amhere
+  if (NOT DEBUG AND UNIX AND X64 AND NOT CMAKE_COMPILER_IS_CLANG)
+    if (${test} MATCHES "common" OR
+        ${test} MATCHES "drcachesim" OR
+        ${test} MATCHES "drcacheoff")
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: OK, I was originally thinking of a per-individual-test attribute, making it easy to add a single test at the test creation site, but I suppose this works for our current purposes.

File: suite/tests/CMakeLists.txt (hunk 2 above)
Comment (derekbruening) [reply suite/tests/CMakeLists.txt]: (Not asking for code changes -- wish github had a checkbox for that.)

File: suite/tests/CMakeLists.txt (hunk 3 above)
Comment (derekbruening) [reply suite/tests/CMakeLists.txt]: (No action required -- wish github had a checkbox for that.)

Comment (hgreving2304) [reply suite/tests/CMakeLists.txt]: You can set the property until after the tests have been added. So this seemed the best place, unless we want to add an parameter that's passed into the torun_ functions.. Ok, thanks.

Comment (derekbruening) [reply suite/tests/CMakeLists.txt]: Like I said, here and in the prior PR comments, "a per-individual-test attribute" which is *ours*.  Just like all the other properties we pass to here using variables named test_foo.

File: suite/tests/CMakeLists.txt (hunk 2 above)
Comment (hgreving2304) [reply suite/tests/CMakeLists.txt]: The newly added tests run about 2s total.

File: suite/tests/CMakeLists.txt (hunk 3 above)
Comment (derekbruening) [reply suite/tests/CMakeLists.txt]: E.g., the _depends attribute, the _dr_ops attribute, the _env attribute, etc.

**Pull request merged**